pandas/tests/test_graphics_others.py

```diff
@@ -96,29 +98,39 @@ def test_hist_layout(self):
     def test_hist_layout_with_by(self):
         df = self.hist_df
 
-        axes = _check_plot_works(df.height.hist, by=df.gender, layout=(2, 1))
+        with tm.assert_produces_warning(UserWarning):
+            axes = _check_plot_works(df.height.hist, by=df.gender,
+                                     layout=(2, 1))
```

Comment:
These are also not raising warnings (with the same comment: don't you get the warnings you are testing for when you pass axes yourself?)